cmake/multi_image.cmake

```diff
@@ -23,125 +23,27 @@ if(IMAGE_NAME)
-  endif()
-endfunction()
-
-function(add_child_image_from_source)
-  # See 'add_child_image'
-  set(oneValueArgs NAME SOURCE_DIR DOMAIN BOARD)
-  cmake_parse_arguments(ACI "" "${oneValueArgs}" "" ${ARGN})
-
-  if (NOT ACI_NAME OR NOT ACI_SOURCE_DIR)
-    message(FATAL_ERROR "Missing parameter, required: NAME SOURCE_DIR")
-  endif()
-
-  # Pass information that the partition manager is enabled to Kconfig.
-  add_overlay_config(
-    ${ACI_NAME}
-    ${NRF_DIR}/subsys/partition_manager/partition_manager_enabled.conf
-    )
-
-  if (${ACI_NAME}_BOARD)
-    message(FATAL_ERROR
-      "${ACI_NAME}_BOARD set in outer scope. Will be ignored, use "
-      "`add_child_image(BOARD ${${ACI_NAME}_BOARD} ...)` for adding a child "
-      "image for specific board")
-  endif()
-
-  # Add the new partition manager domain if needed.
-  # The domain corresponds to the BOARD without the 'ns' suffix.
-  if (ACI_DOMAIN)
-    if ("${ACI_BOARD}" STREQUAL "")
-      message(FATAL_ERROR
-        "No board specified for domain '${ACI_DOMAIN}'. This configuration is "
-        "typically defined in ${BOARD_DIR}/Kconfig")
-    endif()
-
-    set(domain_parent ${${ACI_DOMAIN}_PM_DOMAIN_DYNAMIC_PARTITION})
-    if(DEFINED ${ACI_DOMAIN}_PM_DOMAIN_DYNAMIC_PARTITION
-       AND NOT "${domain_parent}" STREQUAL "${ACI_NAME}"
-    )
-      # A domain may only have one child image, which can then act as a parent
-      # to other images in the domain.
-      # As it is a cache variable we check it's content so that CMake re-run
-      # will pass the check as long as the child image hasn't changed.
-      message(FATAL_ERROR "A domain may only have a single child image."
-        "Current domain image is: ${domain_parent}, `${domain_parent}` is a "
-	"domain parent image, so you may add `${ACI_NAME}` as a child inside "
-	"`${domain_parent}`"
-      )
-    endif()
-    # This needs to be made globally available as it is used in other files.
-    set(${ACI_DOMAIN}_PM_DOMAIN_DYNAMIC_PARTITION ${ACI_NAME} CACHE INTERNAL "")
-
-    if (NOT (${ACI_DOMAIN} IN_LIST PM_DOMAINS))
-      list(APPEND PM_DOMAINS ${ACI_DOMAIN})
-      set_property(GLOBAL APPEND PROPERTY PM_DOMAINS ${ACI_DOMAIN})
-    endif()
-  elseif (NOT ACI_BOARD)
-    # No BOARD is given as argument, this triggers automatic conversion of
-    # *.ns board from parent image.
-    get_board_without_ns_suffix(${BOARD} ACI_BOARD)
-  endif()
-
-  if (NOT ACI_DOMAIN AND DOMAIN)
-    # If no domain is specified, a child image will inherit the domain of
-    # its parent.
-    set(ACI_DOMAIN ${DOMAIN})
-    set(inherited " (inherited)")
+#
+# Function for gathering all CMake arguments relevant to Kconfig and Devicetree 
+# for a child image. The output can be passed directly to a CMake call.
+#
+# Usage:
+#   get_child_image_cmake_args(
+#     CHILD_IMAGE            <name of child image>
+#     OUTPUT_CMAKE_ARGS      <output variable for storing cmake args>
+#   )
+function(get_child_image_cmake_args)
```

Comment:
varargs should not be used when the number of arguments is not variable.

Reply:
ok, will fix